src/asm/CodeGenerator.cpp

```diff
@@ -104,14 +104,23 @@ const FastModificationList<std::unique_ptr<qpu_asm::Instruction>>& CodeGenerator
 
     logging::debug() << "-----" << logging::endl;
     std::size_t index = 0;
-    method.forAllInstructions([&generatedInstructions, &index, &registerMapping, &labelMap](const IntermediateInstruction* instr) -> bool
+
+	IntermediateInstruction* previous = nullptr;
+    method.forAllInstructions([&generatedInstructions, &index, &registerMapping, &labelMap, &previous](const IntermediateInstruction* instr) -> bool
 	{
-    	Instruction* mapped = instr->convertToAsm(registerMapping, labelMap, index);
-		if (mapped != nullptr) {
-			generatedInstructions.emplace_back(mapped);
+		if (instr->mapsToASMInstruction()) {
+			Instruction *mapped = instr->convertToAsm(registerMapping, labelMap, index);
+			if (auto label = dynamic_cast<BranchLabel*>(previous))
+				mapped->previousComment += label->getLabel()->to_string();
```

Comment:
You need to move the assignment of `mapped->previousComment` after checking for `mapped != nullptr`